contracts/earn/strategies/Alpaca/StrategyAlpaca.sol

```diff
@@ -0,0 +1,163 @@
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    /* ----- State Variables ----- */
+
+    address public alpacaVault;
+    address public fairLaunch;
+    uint256 public immutable pid;
+
+    address[] public rewardToWantRoute;
+
+    /* ----- Constant Variables ----- */
+
+    address public constant wrappedEther = address(0xbb4CdB9CBd36B01bD1cBaEBF2De08d9173bc095c);
+    address public constant reward = address(0x8F0528cE5eF7B51152A59745bEfDD91D97091d2F);
+    address public constant uniRouter = address(0x10ED43C718714eb63d5aA57B78B54704E256024E);
+
+    constructor(
+        address initVault,
+        address initAccessManager,
+        address initWant,
+        address initAlpacaVault,
+        address initFairLaunch,
+        uint256 initPid,
+        address[] memory initRewardToWantRoute
+    ) public BaseStrategy(initVault, initAccessManager) {
+        want = initWant;
+        alpacaVault = initAlpacaVault;
+        fairLaunch = initFairLaunch;
+        pid = initPid;
+        rewardToWantRoute = initRewardToWantRoute;
+
+        _giveAllowances();
+    }
+
+    /* ----- Public Functions ----- */
+
+    function harvest() public override whenNotPaused {
+        require(msg.sender == tx.origin || msg.sender == address(vault), 'StrategyAlpaca: EOA_or_vault');
+
+        IFairLaunch(fairLaunch).harvest(pid);
+        uint256 rewardBalance = IERC20(reward).balanceOf(address(this));
+
+        if (rewardBalance > 0) {
+            uint256 wantBalBefore = IERC20(want).balanceOf(address(this));
+            _swapRewards();
+            uint256 wantBalAfter = IERC20(want).balanceOf(address(this));
+            uint256 perfAmount = wantBalAfter.sub(wantBalBefore);
+            chargePerformanceFee(perfAmount);
+            deposit();
+        }
+    }
+
+    function deposit() public override whenNotPaused {
+        uint256 wantBalance = IERC20(want).balanceOf(address(this));
+
+        if (wantBalance > 0) {
+            if (want == wrappedEther) {
+                IWETH(wrappedEther).withdraw(wantBalance);
+                IAlpacaVault(alpacaVault).deposit{value: wantBalance}(wantBalance);
+            } else {
+                IAlpacaVault(alpacaVault).deposit(wantBalance);
+            }
+            IFairLaunch(fairLaunch).deposit(address(this), pid, IAlpacaVault(alpacaVault).balanceOf(address(this)));
+        }
+    }
+
+    function withdraw(uint256 amount) public override {
+        require(msg.sender == address(vault), 'StrategyAlpaca: not_vault');
+
+        uint256 wantBalance = IERC20(want).balanceOf(address(this));
+        if (wantBalance < amount) {
+            uint256 ibAmount = amount.mul(IAlpacaVault(alpacaVault).totalSupply()).div(
+                IAlpacaVault(alpacaVault).totalToken()
+            );
+            IFairLaunch(fairLaunch).withdraw(address(this), pid, ibAmount);
+            IAlpacaVault(alpacaVault).withdraw(IERC20(alpacaVault).balanceOf(address(this)));
```

Comment:
Not a request of change: just want to know why not use 'ibAmount' ?

Reply:
`0x158Da805682BdC8ee32d52833aD41E74bb951E59` Alpaca shares token name like ibUSDT